components/brave_wallet_ui/page/screens/swap/hooks/useSwap.ts

```diff
@@ -488,10 +530,12 @@ export const useSwap = () => {
   const nativeAssetBalance = nativeAsset && getAssetBalance(nativeAsset)
 
   const onClickFlipSwapTokens = useCallback(async () => {
-    setFromToken(toToken)
-    setToToken(fromToken)
+    if (!fromAccount || !fromToken || !toToken || !toAccount) {
+      return
+    }
+    history.push(makeSwapRoute(toToken, toAccount, fromToken, fromAccount))
```

Comment:
The `fromAccount` and `toAccount` don't need to be flipped here. I don't think the user expects that, as `toAccount` can actually be an external address (in the new composer UI).

Let's clear the query params if a flip can lead to a bad state - for example, from and to have different coin types.

Reply:
Updated to your suggestions here!